src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,649 @@
+                                (option) =>
+                                    option.itemType !==
+                                    SelectableOptionMenuItemType.Header
+                            )
+                            .map((option) => option.data || option.text)
+                    );
+            } else {
+                if (onResourceChange)
+                    onResourceChange(
+                        selectedOption?.data || selectedOption?.text,
+                        restOfOptions
+                            ?.filter(
+                                (option) =>
+                                    option.itemType !==
+                                    SelectableOptionMenuItemType.Header
+                            )
+                            .map((option) => option.data || option.text)
+                    );
+            }
+        },
+        [options, selectedOption]
+    );
+
+    const handleOnRender = useCallback(
+        (option: IComboBoxOption) => {
+            return (
+                <div className={classNames.comboBoxOptionWrapper}>
+                    <span className={classNames.comboBoxOptionText}>
+                        {option.text}
+                    </span>
+                    {option.itemType !== SelectableOptionMenuItemType.Header &&
+                        !resourcesState.isLoading &&
+                        resourcesState.adapterResult?.result?.data?.findIndex(
+                            (r) => r.id === option.key
+                        ) === -1 && (
+                            <Icon
+                                iconName="Delete"
+                                aria-hidden="true"
+                                title={t('resourcesPicker.removeFromList')}
+                                style={{ paddingLeft: 12 }}
+                                onClick={(event) => {
+                                    event.stopPropagation();
+                                    handleOnRemove(option);
+                                }}
+                            />
+                        )}
+                </div>
+            );
+        },
+        [classNames, resourcesState]
+    );
+
+    return (
+        <div className={classNames.root}>
+            <VirtualizedComboBox
+                styles={classNames.subComponentStyles.comboBox()}
+                placeholder={placeholder}
+                label={label}
+                options={options}
+                selectedKey={selectedKey}
+                allowFreeform={allowFreeform}
+                autoComplete={'on'}
+                required
+                disabled={disabled}
+                errorMessage={inputError}
+                text={selectedOption?.text} // to show the selectedOption text even there is input error
+                onChange={(_e, option, _idx, value) => {
+                    handleOnChange(option, value);
+                }}
+                onRenderLabel={() => (
```

Comment:
could pull out the custom render for the label to a callback as well if you want.